app/policies/proposal_policy.rb

```diff
@@ -79,9 +79,19 @@ def resolve
       # use subselects instead of left joins to avoid an explicit
       # duplication-removal step
       where_clause = <<-SQL
+        -- requester
         requester_id = :user_id
+        -- approver
         OR EXISTS (SELECT id FROM approvals
                    WHERE proposal_id = proposals.id AND user_id = :user_id)
+        -- delegate
+        OR EXISTS (
+          SELECT approvals.id FROM approvals
+          LEFT OUTER JOIN approval_delegates
+          ON approval_delegates.assigner_id = approvals.user_id
+          WHERE approval_delegates.assignee_id = :user_id
+        )
```

Comment:
It looks like there's not restriction by proposal? I think all that needs to change is to add:
```sql
AND approvals.proposal_id = proposals.id
```